modules/nf-core/bcftools/reheader/main.nf

```diff
@@ -23,15 +26,29 @@ process BCFTOOLS_REHEADER {
     def prefix = task.ext.prefix ?: "${meta.id}"
     def update_sequences = fai ? "-f $fai" : ""
     def new_header       = header ? "-h $header" : ""
+
+    def args2 = task.ext.args2 ?: ''
+    def extension = args2.contains("--output-type b") || args2.contains("-Ob") ? "bcf.gz" :
+                    args2.contains("--output-type u") || args2.contains("-Ou") ? "bcf" :
+                    args2.contains("--output-type z") || args2.contains("-Oz") ? "vcf.gz" :
+                    args2.contains("--output-type v") || args2.contains("-Ov") ? "vcf" :
+                    "vcf.gz"
+
+    def compression = extension.equals("vcf.gz") && !args2.contains("--output-type z") ?
+                    "--output-type z" : ""
```

Comment:
```suggestion
    def args2 = task.ext.args2 ?: '--output-type z'
    def extension = args2.contains("--output-type b") || args2.contains("-Ob") ? "bcf.gz" :
                    args2.contains("--output-type u") || args2.contains("-Ou") ? "bcf" :
                    args2.contains("--output-type z") || args2.contains("-Oz") ? "vcf.gz" :
                    args2.contains("--output-type v") || args2.contains("-Ov") ? "vcf" :
                    "vcf"
```

In most cases the `args2` won't be used by the user. You can simply default it to `--output-type z` to have it emit `vcf.gz` by default

Reply:
ah of course